packages/opentelemetry-resources/test/util/resource-assertions.ts

```diff
@@ -315,6 +319,30 @@ export const assertResource = (
       validations.runtimeDescription
     );
   }
+  if (validations.platform) {
+    assert.strictEqual(
+      resource.attributes['browser.platform'],
+      validations.platform
+    );
+  }
+  if (validations.brands) {
+    assert.strictEqual(
+      (resource.attributes['browser.brands'] as Array<string>).join('.'),
```

Comment:
We can check the value type here first before casting the type.

```suggestion
      assert.ok(Array.isArray(resource.attributes['browser.brands']);
      (resource.attributes['browser.brands'] as string[]).join('.'),
```